lib/cfg/cfg_Converter.ml

```diff
@@ -0,0 +1,79 @@
+open Ast.HeapRegularCommands
+open Cfg_Node
+
+module Converter = struct
+  let rec convert_helper(root: 'a HeapRegularCommand.t) :
```

Comment:
Why is a reference to the same node needed?

Reply:
Where?

Reply:
Because we don't need to pass the whole CFG around, it might be expensive. We need to keep in memory the last node of the CFG otherwise we can't grow the structure.